src/mpl/src/gpu/mpl_gpu_ze.c

```diff
@@ -816,12 +812,34 @@ static int munmapFunction(int nfds, int *fds, void *ptr, size_t size)
     mapped pointers can be from a remote IPC handle
 */
 
-static inline void free_ipc_handle_cache(MPL_ze_ipc_handle_entry_t * cache_entry)
+static inline int free_ipc_handle_cache(MPL_ze_ipc_handle_entry_t * cache_entry)
 {
+    int mpl_err = MPL_SUCCESS;
+    int ret = ZE_RESULT_SUCCESS;
+
     if (cache_entry->mapped_ptr) {
-        munmapFunction(cache_entry->nfds, cache_entry->fds, cache_entry->mapped_ptr,
-                       cache_entry->mapped_size);
+        munmapFunction(cache_entry->nfds, cache_entry->mapped_ptr, cache_entry->mapped_size);
+        if (!cache_entry->handle_cached) {
+            /* free the ipc handle itself */
+            ret = zeMemPutIpcHandle(ze_context, cache_entry->ipc_handle.ipc_handles[0]);
+            ZE_ERR_CHECK(ret);
+        }
     }
+
+    if (cache_entry->handle_cached) {
+        ret = zeMemPutIpcHandle(ze_context, cache_entry->ipc_handle.ipc_handles[0]);
+        ZE_ERR_CHECK(ret);
```

Comment:
Should iterate over `cache_entry->nfds` and close all opened handles

Reply:
done